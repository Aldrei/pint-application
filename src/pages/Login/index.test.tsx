import React from 'react';

import { render } from '@testing-library/react';

jest.mock('../../hooks/useReducerSelector', () => {
  const useAppSelectorBlaBlaBalMocked = jest.fn(() => ({
    status: 'idle',
  }));
  return { useAppSelectorBlaBlaBal: useAppSelectorBlaBlaBalMocked };
});

jest.mock('../../store/hooks', () => {
  const useAppDispatchMocked = jest.fn(() => null);
  return { useAppDispatch: useAppDispatchMocked };
});

import LoginPage from './index';

describe('Login page', () => {
  it('Should be render correctly', () => {
    const nodeEl = render(<LoginPage />);

    const nodeBody = nodeEl.getByTestId('loginContainer');
    const nodeUsername = nodeEl.getByTestId('username');
    const nodePassword = nodeEl.getByTestId('password');
    const nodeButtonLogin = nodeEl.getByTestId('button-login');

    expect(nodeBody).toContainElement(nodeUsername);
    expect(nodeBody).toContainElement(nodePassword);
    expect(nodeBody).toContainElement(nodeButtonLogin);

    expect(nodeEl.baseElement).toMatchSnapshot();
  });

  /**
   * TODO: Test behavior here:
   *        1. Check button is disabled if empty input.
   *        2. If has data:
   *           2.1. Check if make request and redirect to dashboard page.
   *           2.2. Check if make request and render message error.
  */
});
