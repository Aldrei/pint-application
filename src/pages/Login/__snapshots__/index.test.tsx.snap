// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Should be render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-ftvSup iFqLWd"
      data-testid="loginContainer"
    >
      <div
        class="sc-gsnTZi dfpLxU cardComp"
      >
        <input
          class="sc-dkzDqf jpsCuT inputComp"
          data-testid="username"
          name="username"
          placeholder="Email"
        />
        <input
          class="sc-dkzDqf jpsCuT inputComp"
          data-testid="password"
          name="password"
          placeholder="Senha"
          type="password"
        />
        <button
          class="sc-bczRLJ cRnKOA buttonComp"
          color="blue"
          data-testid="button-login"
          disabled=""
        >
          Entrar
        </button>
      </div>
    </div>
  </div>
</body>
`;
